app/models/course.rb

```diff
@@ -148,4 +148,31 @@ def funding
       'fee'
     end
   end
+
+  def publish_sites
+    update_new_to_running
+    update_published_if_running
+    # as update_all is called and doesn't trigger a save this has to be manually tested
+    update(changed_at: Time.now.utc)
+  end
+
+  def publish_enrichment(current_user_id)
+    enrichments.where(status: 'draft')
```

Comment:
potential concurrency issues

change it to make it pick up the latests draft only

```csharp
                var enrichmentPublishRecord = _context.ProviderEnrichments
                    .Where(ie => ie.ProviderCode == providerCode && ie.Status == EnumStatus.Published)
                    .OrderByDescending(x => x.Id)
                    .FirstOrDefault();

```

Reply:
concurrency issues live on the c# side that generates the draft. There is only ever one draft so no issue here.